astropy/coordinates/stokes_coord.py

```diff
@@ -0,0 +1,238 @@
+    # The attributes containing actual information.
+    _represent_as_dict_attrs = {"value"}
+    # Since there is only one attribute, use a column with the name to represent it
+    # (rather than as name.value)
+    _represent_as_dict_primary_data = "value"
+    # Attributes that should be presented as positional arguments to
+    # the class initializer (which takes "stokes" as an argument, not "value").
+    _construct_from_dict_args = ("value",)
+
+    @property
+    def unit(self):
+        return None
+
+    @property
+    def dtype(self):
+        return self._parent._data.dtype
+
+    def new_like(self, cols, length, metadata_conflicts="warn", name=None):
+        """
+        Return a new StokesCoord instance which is consistent with the
+        input ``cols`` and has ``length`` rows.
+
+        This is intended for creating an empty column object whose elements can
+        be set in-place for table operations like join or vstack.
+
+        Parameters
+        ----------
+        cols : list
+            List of input columns
+        length : int
+            Length of the output column object
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts
+        name : str
+            Output column name
+
+        Returns
+        -------
+        col : `~astropy.coordinates.StokesCoord` (or subclass)
+            Empty instance of this class consistent with ``cols``
+
+        """
+        # Get merged info attributes like shape, dtype, format, description, etc.
+        attrs = self.merge_cols_attributes(
+            cols, metadata_conflicts, name, ("meta", "format", "description")
+        )
+
+        # Make an empty StokesCoord.
+        shape = (length,) + attrs.pop("shape")
+        data = np.empty(shape=shape, dtype=dtype)
+        # Get arguments needed to reconstruct class
+        out = self._construct_from_dict(stokes=data)
+
+        # Set remaining info attributes
+        for attr, value in attrs.items():
+            setattr(out.info, attr, value)
+
+        return out
+
+    def get_sortable_arrays(self):
+        """
+        Return a list of arrays which can be lexically sorted to represent
+        the order of the parent column.
+
+        For StokesCoord this is just the underlying values.
+
+        Returns
+        -------
+        arrays : list of ndarray
+        """
+        return [self._parent._data]
+
+
+class StokesCoord(ShapedLikeNDArray):
+    """
+    A representation of stokes coordinates with helpers for converting to profile names.
+
+    Parameters
+    ----------
+    stokes_values : array-like
+        The numeric values representing stokes coordinates.
+    """
+
+    info = StokesCoordInfo()
+
+    def __init__(self, stokes, copy=False):
+        stokes = np.asanyarray(stokes)
+        if stokes.dtype.kind == "U":
+            self._data = self._from_symbols(stokes)
+        else:
+            self._data = stokes
```

Comment:
`_from_symbols` unavoidably makes a copy, but here we could be without a copy even if `copy=True`. So, perhaps rewrite as,
```
data = np.asanyarray(stokes)
if data.dtype.kind == 'U':
    self._data = self._from_symbols(stokes)
else:
    self._data = data.copy() if copy and data is stokes else data
```

Reply:
Note that no validation is done here. I was a bit surprised the data are float. If you generally want that, another option is to just replace what you have here with,
```
    self._data = stokes.astype(float, copy=copy)
```
(Obviously, one could use `int` equally easily.) 

Though if the goal is to avoid copying of data until necessary, then the above is better.

Reply:
So I think that we are likely to have float inputs a lot, as WCS will return floats.

That being said, I think any code which assumes float in this implementation probably comes from the fact it was originally based on Quantity, which is always float.

Reply:
Are we actually requiring an underlying data type anywhere? I think from a consistency perspective rounding to int makes the most logical sense, however, I worry that that will make this even more of a breaking change than it is already as people wont be able to get back to the array they get in the form of a Quantity at the moment.

Reply:
I think I have done this now.